dist/array_extensions.js

```diff
@@ -0,0 +1,44 @@
+(function (Array) {
+	const EACH_ERROR = 'Function Each already exist';
+	const WHERE_ERROR = 'Function Where already exist';
+
+	var extensions = {
+		'each': { method: each, error: EACH_ERROR },
+		'where': { method: where, error:WHERE_ERROR}
+	};
+
+	function each(callback) {
+		var length = this.length;
+		var index = 0;
+
+		while (index < length) {
+			callback.call(null, this[index], index);
+			index++;
+		}
+	}
+
+	function where(callback) {
+		var length = this.length;
+		var index = 0;
+		var newArray = [];
+
+		while (index < length) {
+			let response = callback.call(null, this[index]);
+			if (response === true) {
+				newArray.push(this[index])
+			}
+			index++;
+		}
+		return newArray;
+	}
+
+	for(let key in extensions) {
```

Comment:
Or probably the `extensions` map might be avoided and just have an array with all the extension methods, then you could create and throw just the message you need.